src/Interpreters/Aggregator.cpp

```diff
@@ -1754,8 +1754,8 @@ inline void Aggregator::insertAggregatesIntoColumns(Mapped & mapped, MutableColu
       * It is also tricky, because there are aggregate functions with "-State" modifier.
       * When we call "insertResultInto" for them, they insert a pointer to the state to ColumnAggregateFunction
       *  and ColumnAggregateFunction will take ownership of this state.
-      * So, for aggregate functions with "-State" modifier, the state must not be destroyed
-      *  after it has been transferred to ColumnAggregateFunction.
+      * So, for aggregate functions with "-State" modifier, only states of all combinators that are used
```

Comment:
I didn't understand what this sentence means.

Reply:
For example, if we have function `uniqStateForEachMap` after aggregation we should destroy all states that were created by combinators `-ForEach` and `-Map`, because resulting ColumnAggregateFunction will be responsible only for destruction of the states created by `uniq` function.

Reply:
I will add this example to the comment